log/nsq_aggregator.go

```diff
@@ -0,0 +1,90 @@
+package log
+
+import (
+	"fmt"
+	l "log"
+	"time"
+
+	nsq "github.com/nsqio/go-nsq"
+
+	"github.com/deis/logger/storage"
+)
+
+type nsqAggregator struct {
+	listening bool
+	cfg       *config
+	consumer  *nsq.Consumer
+	handler   nsq.HandlerFunc
+}
+
+func newNSQAggregator(storageAdapter storage.Adapter) Aggregator {
+	return &nsqAggregator{
+		handler: nsq.HandlerFunc(func(msg *nsq.Message) error {
+			if err := handle(msg.Body, storageAdapter); err != nil {
+				msg.Requeue(-1)
+				return newErrNSQHandleFailed(err)
+			}
+			return nil
+		}),
+	}
+}
+
+// Listen starts the aggregator
```

Comment:
No need to make concurrency safe right now - maybe in the future if we want loggers to get more fancy. Also, those docs are 💯 - thanks!